staging/src/k8s.io/apiserver/pkg/storage/interfaces.go

```diff
@@ -236,6 +236,11 @@ type Interface interface {
 
 	// Count returns number of different entries under the key (generally being path prefix).
 	Count(key string) (int64, error)
+
+	// RequestBookmark requests a bookmark to be sent on opened watches.
+	// Note: only watches with matching context grpc metadata as defined by https://github.com/kubernetes/kubernetes/blob/9325a57125e8502941d1b0c7379c4bb80a678d5c/vendor/go.etcd.io/etcd/client/v3/watch.go#L1037-L1042.
+	// Best use the same context that watch was created with.
```

Comment:
Added comment that bookmark will point to the progress of the watch. I used the name "bookmark" to avoid introducing new term to Kubernetes code, but we could use name from etcd "progress notification", which is much clearer.